src/profile/external-identities/FrameworkServiceLink.tsx

```diff
@@ -0,0 +1,37 @@
+import { ClickableProps } from '@terra-ui-packages/components';
+import React, { ReactNode } from 'react';
+import { ButtonPrimary, Link } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { Ajax } from 'src/libs/ajax';
+import { withErrorReporting } from 'src/libs/error';
+import * as Utils from 'src/libs/utils';
+
+interface FrameworkServiceLinkProps extends ClickableProps {
+  linkText: string;
+  providerKey: string;
+  redirectUrl: string;
+  button?: boolean;
+}
+
+export const FrameworkServiceLink = (props: FrameworkServiceLinkProps): ReactNode => {
+  const { linkText, providerKey, redirectUrl, button = false, ...clickableProps } = props;
+  const Component = button ? ButtonPrimary : Link;
+  const style = button ? {} : { display: 'inline-flex', alignItems: 'center' };
+  return (
+    <Component
+      onClick={async () => {
+        const loadAuthUrlAndRedirect = withErrorReporting('Error getting Fence Link', async () => {
+          const result = await Ajax().User.getFenceAuthUrl(providerKey, redirectUrl);
+          window.open(result.url, Utils.newTabLinkProps.target, Utils.newTabLinkProps.rel);
```

Comment:
whats the difference between setHref and window.open?

Reply:
`setHref` was used to populate a `href` property of a link on render. Now, we want to retrieve the authorization _after_ render and dynamically redirect the user.

Reply:
The third argument here happens to work currently because `newTabLinkProps.rel` only contains "noopener". But generally, this argument is not the same as `rel`. They accept different keywords. Also, `rel` takes space separated keywords and this takes comma separated.

- https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel
- https://developer.mozilla.org/en-US/docs/Web/API/Window/open